app/collins/controllers/actions/ipmi/GetPoolsAction.scala

```diff
@@ -0,0 +1,114 @@
+package collins.controllers.actions.ipmi
+
+import scala.concurrent.Future
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+
+import collins.controllers.ResponseData
+import collins.controllers.SecureController
+import collins.controllers.actions.RequestDataHolder
+import collins.controllers.actions.SecureAction
+import collins.controllers.actions.ipaddress.AddressActionHelper
+import collins.models.IpmiInfo
+import collins.models.shared.AddressPool
+import collins.util.security.SecuritySpecification
+
+case class GetPoolsAction(
+  spec: SecuritySpecification,
+  handler: SecureController
+) extends SecureAction(spec, handler) with AddressActionHelper {
+
+  case class ActionDataHolder(pools: Set[String]) extends RequestDataHolder
+
+  override def validate(): Validation = {
+    var pools: Set[String] = {
+      val p = IpmiInfo.AddressConfig.filter(_.poolNames != Set())
```

Comment:
This is much nicer. I thought i tried this originally but I must have had some other error popping up the first time :(